django/db/backends/base/schema.py

```diff
@@ -994,33 +996,70 @@ def _rename_field_sql(self, table, old_field, new_field, new_type):
             "type": new_type,
         }
 
-    def _create_fk_sql(self, model, field, suffix):
-        from_table = model._meta.db_table
-        from_column = field.column
-        _, to_table = split_identifier(field.target_field.model._meta.db_table)
-        to_column = field.target_field.column
+    def _create_constraint_sql(self, table, name, constraint):
+        constraint = Statement(self.sql_constraint, name=name, constraint=constraint)
+        return Statement(self.sql_create_constraint, table=table, constraint=constraint)
 
+    def _create_fk_sql(self, model, field, suffix):
         def create_fk_name(*args, **kwargs):
             return self.quote_name(self._create_index_name(*args, **kwargs))
 
-        return Statement(
-            self.sql_create_fk,
-            table=Table(from_table, self.quote_name),
-            name=ForeignKeyName(from_table, [from_column], to_table, [to_column], suffix, create_fk_name),
-            column=Columns(from_table, [from_column], self.quote_name),
-            to_table=Table(field.target_field.model._meta.db_table, self.quote_name),
-            to_column=Columns(field.target_field.model._meta.db_table, [to_column], self.quote_name),
-            deferrable=self.connection.ops.deferrable_sql(),
+        table = Table(model._meta.db_table, self.quote_name)
+        name = ForeignKeyName(
+            model._meta.db_table,
+            [field.column],
+            split_identifier(field.target_field.model._meta.db_table)[1],
+            [field.target_field.column],
+            suffix,
+            create_fk_name,
         )
+        column = Columns(model._meta.db_table, [field.column], self.quote_name)
+        to_table = Table(field.target_field.model._meta.db_table, self.quote_name)
+        to_column = Columns(field.target_field.model._meta.db_table, [field.target_field.column], self.quote_name)
+        deferrable = self.connection.ops.deferrable_sql()
+        if self.sql_create_fk:
+            # XXX: Not sure if we should deprecate this path? The foreign key
+            # creation syntax is quite standard and none of builtin backends
+            # override this.
```

Comment:
I guess this will have to be addressed. There seems to be [quite a bit of overrides of `sql_create_fk`](https://github.com/search?l=Python&q=sql_create_fk&type=Code) so this will have to go through a deprecation period at least.

Reply:
Looking at the diff for the base SQL templates with stable/2.1.x, I think we may also need to be more careful with `sql_create_check` and `sql_delete_check`, as they are completely removed. They were added in 2012, even though the main check constraints feature is new in 2.2.

```diff
-    sql_create_check = "ALTER TABLE %(table)s ADD CONSTRAINT %(name)s CHECK (%(che
ck)s)"
-    sql_delete_check = "ALTER TABLE %(table)s DROP CONSTRAINT %(name)s"
+    sql_foreign_key_constraint = "FOREIGN KEY (%(column)s) REFERENCES %(to_table)s
 (%(to_column)s)%(deferrable)s"
+    sql_unique_constraint = "UNIQUE (%(columns)s)"
+    sql_check_constraint = "CHECK (%(check)s)"
+    sql_create_constraint = "ALTER TABLE %(table)s ADD %(constraint)s"
+    sql_delete_constraint = "ALTER TABLE %(table)s DROP CONSTRAINT %(name)s"
+    sql_constraint = "CONSTRAINT %(name)s %(constraint)s"
 
-    sql_create_unique = "ALTER TABLE %(table)s ADD CONSTRAINT %(name)s UNIQUE (%(c
olumns)s)"
-    sql_delete_unique = "ALTER TABLE %(table)s DROP CONSTRAINT %(name)s"
+    sql_create_unique = None
+    sql_delete_unique = sql_delete_constraint
 
-    sql_create_fk = (
-        "ALTER TABLE %(table)s ADD CONSTRAINT %(name)s FOREIGN KEY (%(column)s) "
-        "REFERENCES %(to_table)s (%(to_column)s)%(deferrable)s"
-    )
+    sql_create_fk = None
     sql_create_inline_fk = None
-    sql_delete_fk = "ALTER TABLE %(table)s DROP CONSTRAINT %(name)s"
+    sql_delete_fk = sql_delete_constraint
 
     sql_create_index = "CREATE INDEX %(name)s ON %(table)s (%(columns)s)%(extra)s"
     sql_delete_index = "DROP INDEX %(name)s"
 
     sql_create_pk = "ALTER TABLE %(table)s ADD CONSTRAINT %(name)s PRIMARY KEY (%(
columns)s)"
-    sql_delete_pk = "ALTER TABLE %(table)s DROP CONSTRAINT %(name)s"
+    sql_delete_pk = sql_delete_constraint
```

Reply:
Right, I guess we could keep them around behind a `@property` and warn on access?